app/forms/curation_concerns/forms/collection_edit_form.rb

```diff
@@ -12,6 +12,31 @@ class CollectionEditForm
       def required?(key)
         model_class.validators_on(key).any? { |v| v.is_a? ActiveModel::Validations::PresenceValidator }
       end
+
+      # @return [Hash] All generic files in the collection, file.to_s is the key, file.id is the value
+      def select_files
+        Hash[
+          all_files
+        ]
+      end
+
+      private
+
+        def all_files
+          member_presenters.flat_map(&:file_presenters).map { |x| [x.to_s, x.id] }
```

Comment:
select_files needs to get the titles of all the files. There isn't a `to_s` method on generic_files (which returns the filename if there isn't a title assigned). The code is refactored to access the files through presenters.